builds/checkin/dotnet.yaml

```diff
@@ -2,111 +2,106 @@ trigger: none
-            echo "Detected changes outside of edgelet, docs and mqtt folders"
-            echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
-          fi
-        displayName: Check changes in sources
-        name: check_files
+  displayName: Check pipeline preconditions (changes ARE NOT in either edgelet, docs, or mqtt folder)
+  pool:
+    name: $(pool.linux.name)
+    demands:
+    - ImageOverride -equals agent-aziotedge-ubuntu-20.04-msmoby
+  steps:
+  - bash: |
+      git log -m -1 --name-only --first-parent --pretty="" | egrep -i -v '^(edgelet|doc|mqtt)'
+      if [[ $? == 0 ]]; then
+        echo "Detected changes outside of edgelet, docs and mqtt folders"
+        echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
+      fi
+    displayName: Check changes in sources
+    name: check_files
 
 ################################################################################
-  - job: linux_amd64
+- job: linux_amd64
 ################################################################################
-    displayName: Linux amd64
-    dependsOn: check_run_pipeline
-    condition: eq(dependencies.check_run_pipeline.outputs['check_files.RUN_PIPELINE'], 'true')
-    pool:
-      vmImage: "ubuntu-20.04"
-    steps:
-      - task: Bash@3
-        displayName: Install Prerequisites
-        inputs:
-          filePath: scripts/linux/installPrereqs.sh
-      - task: Bash@3
-        displayName: Build
-        inputs:
-          filePath: scripts/linux/buildBranch.sh
-          arguments: -c "$(configuration)"
-      - task: Bash@3
-        displayName: Test
-        inputs:
-          filePath: scripts/linux/runTests.sh
-          arguments: '"--filter Category=Unit&Category!=GetLogsTests"'
-      - task: Bash@3
-        displayName: Run GetLogsTests Tests
-        inputs:
-          filePath: scripts/linux/runTests.sh
-          arguments: '"--filter Category=GetLogsTests"'
-      - task: PublishTestResults@2
-        displayName: Publish Test Results
-        inputs:
-          testRunner: VSTest
-          testResultsFiles: "**/TestResults/*.trx"
-        condition: succeededOrFailed()
+  displayName: Linux amd64
+  dependsOn: check_run_pipeline
+  condition: eq(dependencies.check_run_pipeline.outputs['check_files.RUN_PIPELINE'], 'true')
+  pool:
+    name: $(pool.linux.name)
+    demands:
+    - ImageOverride -equals agent-aziotedge-ubuntu-20.04-msmoby
+  steps:
+  - script: scripts/linux/installPrereqs.sh
+    displayName: Install Prerequisites
+  - script: scripts/linux/buildBranch.sh -c '$(configuration)'
+    displayName: Build
+  - script: scripts/linux/runTests.sh '--filter Category=Unit&Category!=GetLogsTests'
+    displayName: Test
+  - script: scripts/linux/runTests.sh '--filter Category=GetLogsTests'
+    displayName: Run GetLogsTests Tests
+  - task: PublishTestResults@2
+    displayName: Publish Test Results
+    inputs:
+      testRunner: VSTest
+      testResultsFiles: "**/TestResults/*.trx"
```

Comment:
should this be a `'` not `"`?

Reply:
Yes I think it can be single quotes. Updating...

Reply:
Fixed.